src/nbl/video/CVulkanQueue.cpp

```diff
@@ -0,0 +1,111 @@
+#include "nbl/video/CVulkanQueue.h"
+
+#include "nbl/video/CVKLogicalDevice.h"
+#include "nbl/video/CVulkanFence.h"
+#include "nbl/video/CVulkanSemaphore.h"
+#include "nbl/video/CVulkanPrimaryCommandBuffer.h"
+
+namespace nbl {
+namespace video
+{
+
+CVulkanQueue::CVulkanQueue(CVKLogicalDevice* vkdev, VkQueue vkq, uint32_t _famIx, E_CREATE_FLAGS _flags, float _priority) : 
+    IGPUQueue(_famIx, _flags, _priority), m_vkdevice(vkdev), m_vkqueue(vkq)
+{
+    
+}
+
+void CVulkanQueue::submit(uint32_t _count, const SSubmitInfo* _submits, IGPUFence* _fence)
+{
+    auto vkdev = m_vkdevice->getInternalObject();
+    auto* vk = m_vkdevice->getFunctionTable();
+
+    uint32_t waitSemCnt = 0u;
+    uint32_t signalSemCnt = 0u;
+    uint32_t cmdBufCnt = 0u;
+
+    for (uint32_t i = 0u; i < _count; ++i)
+    {
+        const auto& sb = _submits[i];
+        waitSemCnt += sb.waitSemaphoreCount;
+        signalSemCnt += sb.signalSemaphoreCount;
+        cmdBufCnt += sb.commandBufferCount;
+    }
+
+    constexpr uint32_t STACK_MEM_SIZE = 1u<<15;
+    uint8_t stackmem_[STACK_MEM_SIZE]{};
+    uint8_t* mem = stackmem_;
+    uint32_t memSize = STACK_MEM_SIZE;
+
+    const uint32_t submitsSz = sizeof(VkSubmitInfo)*_count;
+    const uint32_t memNeeded = submitsSz + (waitSemCnt + signalSemCnt)*sizeof(VkSemaphore) + cmdBufCnt*sizeof(VkCommandBuffer);
+    if (memNeeded > memSize)
+    {
+        memSize = memNeeded;
+        mem = reinterpret_cast<uint8_t*>( _NBL_ALIGNED_MALLOC(memSize, _NBL_SIMD_ALIGNMENT) );
+    }
+
+    auto raii_ = core::makeRAIIExiter([mem,stackmem_]{ _NBL_ALIGNED_FREE(mem); });
```

Comment:
we should wrap this up in something, since its a very common trick

maybe something a-la `dynamic_array` but that's the converse (not creatable on the heap, only on the stack, and does a RAII exit)

```cpp
template<typename T, size_t stackSize=1u<<15u>
class stack_dynamic_array
```

Reply:
actually for even more versatility
```cpp
template<typename T, size_t stackSize=1u<<15u, template<typename> allocator>
class stack_dynamic_array
```
then use `allocator<T>` to allocate and free the memory (and use it statefully C++11 style, not crappy C++03)

Reply:
did this get done?

Reply:
I'll do it